### DISCUSSION TITLE
Question about Visualization and Object Constraints

### DISCUSSION THREAD

**Discussion by linhy97** [NVlabs/curobo#346] [Software Q&A] [answered]
Thank you for your contributions to supply an excellent motion planning package.
After installing the environment, i have two questions, hope for your help:

i) is there any visualization module we can use directly, such as plotting the trajectories in a simulation environment. I think store the trajectories and word config to files, and parse them in pybullet engine is complex.
ii) In my situation, i need to control the ur10 robot to grasp some cube, in pybullet i will construct a fixed constraint relationship, and take them as a whole into account. Thus, i want to know if there exists any method to consider ur10 robot and the things it grasp as a whole?

Hope for you answers~

**Comment by qingquluofeng** [comment]
@linhy97 Hello, have you solved this problem? I don't know how to consider the robot arm and the object it grabs as a whole for path planning in pybullet.

**Comment by balakumar-s** [comment, marked as answer]
We provide an animated usd export option that generates a usd file containing the robot trajectory. This can be opened with a USD viewer (Isaac Sim, Blender, USDView, native viewer in Mac). Here is an example: https://github.com/NVlabs/curobo/blob/main/examples/usd_example.py

To attach an object to the gripper, see this example: https://curobo.org/advanced_examples/2_block_stacking_example.html

**Reply by qingquluofeng** [reply]
Thank you for your answers !
